src/transformers/modeling_utils.py

```diff
@@ -1371,11 +1373,53 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                             else:
                                 raise ValueError from e
                     except (UnicodeDecodeError, ValueError):
-                        raise OSError(
-                            f"Unable to load weights from pytorch checkpoint file for '{pretrained_model_name_or_path}' "
-                            f"at '{resolved_archive_file}'. "
-                            "If you tried to load a PyTorch model from a TF 2.0 checkpoint, please set from_tf=True."
-                        )
+                        # Check sha256
+                        def sha256(filename, block_size=4096 * 1024):
+                            # Python program to find SHA256 hash string of a file
+
+                            sha256_hash = hashlib.sha256()
+                            with open(filename, "rb") as f:
+                                # Read and update hash string value in blocks of 4Mo
+                                for byte_block in iter(lambda: f.read(block_size), b""):
+                                    sha256_hash.update(byte_block)
+
+                            return sha256_hash.hexdigest()
+
+                        actual_sha = sha256(resolved_archive_file)
+                        metafilename = f"{resolved_archive_file}.json"
+                        try:
+                            with open(metafilename, "r", encoding="utf-8") as f:
+                                data = json.load(f)
+                                # For some reason etag contains quotes, we just want the sha.
+                                expected_sha = data["etag"][1:-1]
+                        except Exception:
+                            expected_sha = None
```

Comment:
All of this should probably be an util function to be re-used for everywhere we download files.

Reply:
+1

Reply:
Actually I am not sure we can do that for **every** file.

As @julien-c mentionned, the Etag validation is only available for LFS files.

We can definitely organise this code a bit better.